docs/api/v3.rst

```diff
@@ -0,0 +1,727 @@
+                    "fail_on_warning": false
+                },
+                "submodules": {
+                    "include": "all",
+                    "exclude": [],
+                    "recursive": true
+                }
+            },
+            "links": {
+                "self": "/api/v3/projects/pip/builds/8592686/",
+                "project": "/api/v3/projects/pip/",
+                "version": "/api/v3/projects/pip/versions/latest/",
+                "commands": "/api/v3/projects/pip/builds/8592686/commands/"
+            }
+        }
+
+    :>json integer id: The ID of the build
+    :>json string date: The ISO-8601 datetime of the build.
+    :>json integer duration: The length of the build in seconds.
+    :>json string state: The state of the build (one of ``triggered``, ``building``, ``installing``, ``cloning``, or ``finished``)
+    :>json boolean success: Whether the build was successful
+    :>json string error: An error message if the build was unsuccessful
+    :>json string commit: A version control identifier for this build (eg. the commit hash)
+    :>json string builder: The hostname server that built the docs
+    :>json string cold_storage: Whether the build was removed from database and stored externally
+
+    :query boolean include_config: whether or not include the configs used for this build. Default is ``false``
+
+    :requestheader Authorization: optional token to authenticate.
+
+    :statuscode 200: Success
+    :statuscode 404: There is no ``Build`` with this ID
+
+
+.. http:get:: /api/v3/projects/(str:project_slug)/builds/latest/
+
+    Retrieve details for latest build on this project.
+
+    **Example request**:
+
+    .. sourcecode:: bash
+
+        $ curl https://readthedocs.org/api/v3/projects/pip/builds/latest/
+
+    **Example response**:
+
+    `See Build details <#build-details>`_
+
+    :requestheader Authorization: optional token to authenticate.
+
+
+Builds listing
+++++++++++++++
+
+.. http:get:: /api/v3/projects/(str:project_slug)/builds/
+
+    Retrieve list of all the builds on this project.
+
+    **Example request**:
+
+    .. sourcecode:: bash
+
+        $ curl https://readthedocs.org/api/v3/projects/pip/builds/
+
+    **Example response**:
+
+    .. sourcecode:: json
+
+        {
+            "count": 15,
+            "next": "/api/v3/projects/pip/builds?limit=10&offset=10",
+            "previous": null,
+            "results": ["BUILD"]
+        }
+
+    :query string commit: commit hash to filter the builds returned by commit
+    :query boolean running: whether or not to filter the builds returned by currently building
+
+    :requestheader Authorization: optional token to authenticate.
+
+
+Build triggering
+++++++++++++++++
+
+
+.. http:post:: /api/v3/projects/(string:project_slug)/builds/
+
+    Trigger a new build for this project.
+
+    **Example request**:
+
+    .. sourcecode:: json
+
+        {
+            "version": "latest",
```

Comment:
I'm doubting about this now.

Instead of sending an argument here, we could do:

* `/api/v3/projects/(string:project_slug)/builds/` for triggering the default version configured (usually `latest`)
* `/api/v3/projects/(string:project_slug)/versions/(string:version_slug)/builds/` to trigger the specific version

This approach seems more clear and follow the pattern we are using, I guess.

Reply:
`Build` is a weird one, the global pk communicates that this is a top-level object, but it effectively is not in any application. I see a use case for the first endpoint in the case of querying a build, but in the case of triggering a build, i'd definitely vote for the second option. In the case of querying a build, we don't necessarily care what the version is, but it might still help to be explicit about that relationship in our API modeling -- at least just for consistency?

Reply:
Let's move this to a new issue though, the conversation here has dragged this PR out.

Reply:
> In the case of querying a build, we don't necessarily care what the version is, but it might still help to be explicit about that relationship in our API modeling -- at least just for consistency?

This was where I was doubting also. I like consistency but maybe that purity complicates the usage from a user perspective. Although, if we provide good links on the responses with the related object this shouldn't be a problem.

Reply:
That's true too, yeah. Maybe also worth considering, long term, we might want to change the Build model pk to a compound key, in which case these two would be different builds:

* `/api/v3/projects/foo/versions/latest/build/1`
* `/api/v3/projects/foo/versions/bar/build/1`

The fact that we expose build pk as a global pk has always been sort of odd.

Reply:
I like this idea.

Do we need to "modify the Build model pk"? Can't we make this an API identifier only for now?

* `1` would be the first (sorted by date) Build for this version
* `3` would be the third one
* `-1` would be the last one

Internally, we just sort the queryset (by `date` or `-date`) and pick the `N` element.

What do you think?